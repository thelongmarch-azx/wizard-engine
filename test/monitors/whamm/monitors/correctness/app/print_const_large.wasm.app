../../apps/i32_const_large.wasm
