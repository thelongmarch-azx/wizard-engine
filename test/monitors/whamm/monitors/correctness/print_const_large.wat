;; A correctly implemented whamm probe which prints out all `i32.const`
(module
  (import "wizeng" "puti" (func $puti (param i32)))
  (import "wizeng" "puts" (func $puts (param i32 i32)))

  (export "wasm:opcode:i32.const/$large(imm0)/(fid, pc, imm0)" (func $print_const))
  (export "" (memory $mem))

  (memory $mem 1)

  (data (i32.const 0x00) "func=")
  (data (i32.const 0x10) ", pc=")
  (data (i32.const 0x20) ": ")
  (data (i32.const 0x30) "\n")

  (func (export "$large") (param $imm0 i32) (result i32)
    (i32.ge_s (local.get $imm0) (i32.const 100))
  )

  (func $print_const (param $func i32) (param $pc i32) (param $num i32)
    (call $puts (i32.const 0x00) (i32.const 5))
    (call $puti (local.get $func))
    (call $puts (i32.const 0x10) (i32.const 5))
    (call $puti (local.get $pc))
    (call $puts (i32.const 0x20) (i32.const 2))
    (call $puti (local.get $num))
    (call $puts (i32.const 0x30) (i32.const 1))
  )
)
