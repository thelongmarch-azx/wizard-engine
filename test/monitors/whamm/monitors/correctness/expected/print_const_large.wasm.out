func=0, pc=8: 100
func=0, pc=13: 101
func=0, pc=18: 102
